pkg/apis/core/types.go

```diff
@@ -2568,9 +2568,10 @@ type PodSpec struct {
 	// file if specified. This is only valid for non-hostNetwork pods.
 	// +optional
 	HostAliases []HostAlias
-	// If specified, indicates the pod's priority. "SYSTEM" is a special keyword
-	// which indicates the highest priority. Any other name must be defined by
-	// creating a PriorityClass object with that name.
+	// If specified, indicates the pod's priority. "system-node-critical" and
+	// "system-cluster-critical" are two special keywords which indicate the
+	// highest priorities with the former being the highest priority. Any other
+	// name must be defined by creating a PriorityClass object with that name.
```

Comment:
Does this mean creating a PriorityClass with those two names will be forbidden? Will there ever be additional reserved keyword priorities? Should we reserve the `system-` prefix and disallow priority class objects with that prefix to allow us to expand predefined ones in the future?

Reply:
> Does this mean creating a PriorityClass with those two names will be forbidden?

Yes, Priority admission controller enforces this.

> Will there ever be additional reserved keyword priorities?

There is no plan to add more, but we might add if there is REAL need for a reserved name.

> Should we reserve the system- prefix and disallow priority class objects with that prefix to allow us to expand predefined ones in the future?

That's actually a good suggestion. We have already reserved all priority values >2 billion for system critical pods. We may as well want to reserved the "system-" prefix for such PriorityClass names. I will send a PR.

Reply:
@liggitt, is there anything else I can answer or address in this PR?

Reply:
Same doc changes need to be made in the v1 types go file, and generated docs updated. Otherwise, no one will see this information

Reply:
Of course! I forgot to do that. It is done now. Thanks!